File: backend/src/modules/aluno_aula/entity/AlunoAula.js
class AlunoAula {
  constructor({ data_inscricao, status, presenca, aluno_id_aluno, aula_id_aula }) {
    this.data_inscricao = data_inscricao;
    this.status = status;
    this.presenca = presenca;
    this.aluno_id_aluno = aluno_id_aluno;
    this.aula_id_aula = aula_id_aula;
  }
  validate() {
    const errors = [];
    if (!this.aluno_id_aluno) errors.push('Aluno é obrigatório');
    if (!this.aula_id_aula) errors.push('Aula é obrigatória');
    return errors;
  }
}
module.exports = AlunoAula;

File: backend/src/main/java/com/vitalativa/entity/AlunoAula.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import java.time.LocalDate;

@Entity
@Table(name = "aluno_aula")
public class AlunoAula {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_aluno_aula")
    private Integer idAlunoAula;
    
    @Column(name = "data_inscricao")
    private LocalDate dataInscricao;
    
    private String status;
    private Boolean presenca;
    
    @NotNull(message = "Aluno é obrigatório")
    @Column(name = "aluno_id_aluno")
    private Integer alunoIdAluno;
    
    @NotNull(message = "Aula é obrigatória")
    @Column(name = "aula_id_aula")
    private Integer aulaIdAula;
    
    public AlunoAula() {}
    
    public Integer getIdAlunoAula() { return idAlunoAula; }
    public void setIdAlunoAula(Integer idAlunoAula) { this.idAlunoAula = idAlunoAula; }
    public LocalDate getDataInscricao() { return dataInscricao; }
    public void setDataInscricao(LocalDate dataInscricao) { this.dataInscricao = dataInscricao; }
    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }
    public Boolean getPresenca() { return presenca; }
    public void setPresenca(Boolean presenca) { this.presenca = presenca; }
    public Integer getAlunoIdAluno() { return alunoIdAluno; }
    public void setAlunoIdAluno(Integer alunoIdAluno) { this.alunoIdAluno = alunoIdAluno; }
    public Integer getAulaIdAula() { return aulaIdAula; }
    public void setAulaIdAula(Integer aulaIdAula) { this.aulaIdAula = aulaIdAula; }
}